### DISCUSSION TITLE
Downloading a ZIP file

### DISCUSSION THREAD

**Discussion by jipenaflor** [Kitware/trame#300] [Q&A] [answered]
Hi! I am making a simulation application using OpenFOAM, and I would like to enable the user to download the results of their simulation in .zip file. How would I do it?

I tried following the example [here](https://github.com/Kitware/trame/blob/master/examples/validation/core/24_vtk_download_image.py), but it seems that the return method require an input of type byte.
```
@ctrl.trigger("download_results")
def download_results():
        archived = shutil.make_archive('user', 'zip', user_directory)
        return self._server.protocol.addAttachment(archived)

vuetify.VBtn(
        "Download Results",
         click="utils.download('user.zip', trigger('download_results'), 'zip')",
         variant="tonal",
         classes="pb-1"
         )
```

**Comment by jourdain** [comment, marked as answer]
Read the zip file and send the bytes over.

**Reply by jipenaflor** [reply]
Got it. Thanks!

**Comment by jourdain** [comment]
If you want to serve it over http, you will have to serve the directory (you can create a module to do that) that will contain your zip and dynamically defined the url to download.
